app/components/provider_interface/conditions_component.rb

```diff
@@ -26,6 +26,8 @@ def application_state
     end
 
     def conditions_met?
+      return false if application_choice.status_before_deferral&.eql?('pending_conditions')
```

Comment:
This is much better but how about also checking that current status is 'deferred'? This is because `status_before_deferral` is not reset when offers are re-instated and such offers will run into similar inconsistencies.

Reply:
A candidate may have satisfied offer conditions before the deferral, in which case `status_before_deferral` will be `recruited`, but the provider may choose to re-instate the offer as `pending_conditions` because the candidate may require e.g. a new DBS check. In this case, the current code will actually return the right thing, but this feels slightly accidental.

In the somewhat less likely scenario of a `pending_conditions` state before the deferral but a `recruited` state when the provider reinstates the offer, I'd expect the current code to say `pending_conditions`.

Reply:
Ah yes, good point, I didn't spot that the `status_before_deferral` field was not unset/reset when an offer was reconfirmed.
I've amended here https://github.com/DFE-Digital/apply-for-teacher-training/pull/3962/commits/a0db628d2a887385abfc964e55b7e7c3f9488193 along with an additional test.